listener/pkg/v2/types/event.go

```diff
@@ -0,0 +1,11 @@
+package types
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type WatchEvent struct {
+	Owner      ObjectKey               `json:"owner"`
```

Comment:
Since we are anyways using the `k8s.io/apimachinery` dependency, can't we use its `types.NamespacedName` here for `Owner` and `Watched`?
The previously used`client.ObjectKey` anyways was just an alias for it.